src/main/kotlin/interpreter/stack/StackFrameInfo.kt

```diff
@@ -57,30 +49,32 @@ abstract class StackFrameInfo {
         override val filename: PyString
             get() = PyString(frame.function.code.filename)
 
-        override val hasDisassembly: Boolean = true
-        override val hasStack: Boolean = true
-
-        override fun getDisassembly(): String {
-            return this.frame.function.code.getDisassembly(this.frame)
-        }
-
-        override fun getStack(): Deque<PyObject> {
-            return this.frame.stack
-        }
-
-        override fun getTracebackString(): String {
-            val module = this.frame.function.module
-            val sourceFile = module.path
-            val sourceLines = Files.readAllLines(sourceFile)
-            val lineNo = this.frame.getLineNo()
-            val sourceLine = sourceLines[lineNo].trimIndent()
-
-            return "File ${frame.function.code.filename}, " +
-                    "instruction idx ${frame.bytecodePointer}, " +
-                    "line ${frame.getLineNo()}, " +
-                    "in ${frame.function.code.codename}\n" +
-                    "    $sourceLine"
-        }
+        override val disassembly: String
+            get() {
```

Comment:
gets like these can all be one line: `get() = frame.function.code.getDisassembly(frame)`